resources/views/livewire/slack-settings-form.blade.php

```diff
@@ -0,0 +1,222 @@
+                    //    'Content-Type': 'application/json',
+                },
+                data: {
+                    'slack_endpoint': $('#slack_endpoint').val(),
+                    'slack_channel': $('#slack_channel').val(),
+                    'slack_botname': $('#slack_botname').val(),
+                },
+                dataType: 'json',
+                accepts: {
+                    text: "application/json"
+                },
+                success: function (data) {
+                    $('#save_slack').removeAttr('disabled');
+                    $("#slacktesticon").html('');
+                    $("#slacktestrow").addClass('text-success');
+                    $("#slackteststatus").addClass('text-success');
+                    //TODO: This is a bit hacky...Might need some cleanup
+                    $("#slackteststatus").html('<i class="fas fa-check text-success"></i> {{ trans('admin/settings/message.slack.success_pt1') }} ' + $('#slack_channel').val() + '{{ trans('admin/settings/message.slack.success_pt2') }}');
+                },
+                error: function (data) {
+                    if (data.responseJSON) {
+                        var errors = data.responseJSON.errors;
+                        var error_msg = data.responseJSON.message;
+                    } else {
+                        var errors;
+                        var error_msg = trans('admin/settings/message.slack.error');
+                    }
+                    var error_text = '';
+                    $('#save_slack').attr("disabled", true);
+                    $("#slacktesticon").html('');
+                    $("#slackteststatus").addClass('text-danger');
+                    $("#slacktesticon").html('<i class="fas fa-exclamation-triangle text-danger"></i><span class="text-danger">' + error_msg+ '</span>');
+
+                    if (data.status == 500) {
+                        $('#slackteststatus').html('{{  trans('admin/settings/message.slack.500') }}');
+                    } else if ((data.status == 400) || (data.status == 422)) {
+                        console.log('Type of errors is '+ typeof errors);
+                        console.log('Data status was 400 or 422');
+                        if (typeof errors != 'string') {
+
+                            console.log(errors.length);
+                            for (i in errors) {
+                                if (errors[i]) {
+                                    error_text += '<li>Error: ' + errors[i];
+                                }
+                            }
+                        } else {
+                            error_text = errors;
+                        }
+                        $('#slackteststatus').html(error_text);
+                    } else {
+                        $('#slackteststatus').html(data.responseText.message);
+                    }
+                }
+            });
+            return false;
+        });
+    </script>
```

Comment:
So as much as possible of this JS chunk should probably get moved somewhere into the Livewire controller, so it's PHP and not as much JS. Generally, for Livewire stuff, the JS should be just short chunks of glue between whatever-library-we're-talking-about and Livewire (ideally none, but that's not always practical). Then we should have as much code as possible living in the Livewire controller, as we can reasonably do.